src/components/Input/BaseInput/UncontrolledBaseInput.tsx

```diff
@@ -1,19 +1,26 @@
 import React, {useState, ChangeEvent} from 'react';
 import type {UncontrolledBaseInputProps} from './BaseInput.types';
 import {ControlledBaseInput} from './ControlledBaseInput';
+import {filterInputValue} from '~/utils/filterInputValue';
 
-export const UncontrolledBaseInput: React.FC<UncontrolledBaseInputProps> = ({defaultValue, onChange, ...props}) => {
+export const UncontrolledBaseInput = React.forwardRef<HTMLInputElement, UncontrolledBaseInputProps>(({defaultValue, onChange, hasFilteredValue = false, allowNegative, allowDecimal, separator, ...props}, ref) => {
     const [inputValue, setBaseInputValue] = useState(defaultValue);
 
     const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
-        setBaseInputValue(event.target.value);
+        let value = event.target.value;
+
+        if (hasFilteredValue) {
+            value = filterInputValue(event.target.value, allowNegative, allowDecimal, separator);
+        }
```

Comment:
Instead of using `hasFilteredValue` to identify if you need to filter the value, hardcode the filtering into the function. I suggest replacing this prop with `filteredFunction` to pass the function to the component. 
In that case, we will be more future-proof to pass a custom filtered function.

In that case, don't forget to update omit types.